### ISSUE TITLE
tests beancount.prices.sources.oanda_test.TestOandaGetHistorical fail because of timezone

### ISSUE THREAD

**Issue by blais** [beancount/beancount#265]
**[Original report](https://bitbucket.org/blais/beancount/issue/265) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

I get the following failures:

```
ERROR: test_valid_after (beancount.prices.sources.oanda_test.TestOandaGetHistorical)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 164, in test_valid_after
    D('1.3100'))
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 139, in _check_valid
    srcprice = self.fetcher.get_historical_price('USD_CAD', query_date)
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda.py", line 163, in get_historical_price
    for item in time_prices
ValueError: min() arg is an empty sequence

======================================================================
FAIL: test_valid_same_date (beancount.prices.sources.oanda_test.TestOandaGetHistorical)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 150, in test_valid_same_date
    D('1.4100'))
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 141, in _check_valid
    srcprice)
AssertionError: Sourc[19 chars]l('1.4100'), time=datetime.datetime(2017, 1, 2[45 chars]CAD') != Sourc[19 chars]l('1.3900'), time=datetime.datetime(2017, 1, 2[42 chars]CAD')

======================================================================
FAIL: test_valid_before (beancount.prices.sources.oanda_test.TestOandaGetHistorical)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 157, in test_valid_before
    D('1.4700'))
  File "/home/tbm/scratch/cvs/beancount/beancount/prices/sources/oanda_test.py", line 141, in _check_valid
    srcprice)
AssertionError: Sourc[20 chars]('1.4700'), time=datetime.datetime(2017, 1, 23[44 chars]CAD') != Sourc[20 chars]('1.4500'), time=datetime.datetime(2017, 1, 23[41 chars]CAD')

----------------------------------------------------------------------
Ran 5 tests in 0.004s

FAILED (errors=1, failures=2)
```

However, with the following change it works for me:

```
-        start_utc = datetime.datetime(date.year, date.month, date.day, 0, 0, 0,
+        start_utc = datetime.datetime(date.year, date.month, date.day, 1, 0, 1,
```

Note that I'm in UTC+2 right now.

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

`test_valid_after` fails with:

```
ValueError: min() arg is an empty sequence
```
we have:

```
Start: 2017-01-19 23:00:00+00:00
End: 2017-01-20 23:00:00+00:00
```

values (`time_prices`)

```
2017-01-21 00:00:00+00:00 1.3100
2017-01-21 08:00:00+00:00 1.3300
2017-01-21 16:00:00+00:00 1.3500
2017-01-22 00:00:00+00:00 1.3700
2017-01-22 08:00:00+00:00 1.3900
2017-01-22 16:00:00+00:00 1.4100
2017-01-23 00:00:00+00:00 1.4300
2017-01-23 08:00:00+00:00 1.4500
2017-01-23 16:00:00+00:00 1.4700
```

this is empty since start_time is smaller than all dates we have in `time_prices`:

```
            before_time, before_price = min(item
                                            for item in time_prices
                                            if item[0] < start_utc)
```

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Hi Michael,
I've managed to reproduce this test (I'd never written tests with temporary remote timezones, this is an interesting learning experience). 
I'll address this shortly.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #265: Refined the timezone specification of the price sources and fixed corresponding unit tests failures.
